private/plan/plan.go

```diff
@@ -23,17 +25,22 @@ func Resources(ctx context.Context, conf *evaluator.Conf, idx compile.Index, inp
 		return nil, err
 	}
 
-	rt := ruletable.NewProtoRuletable()
+	protoRT := ruletable.NewProtoRuletable()
 
-	if err := ruletable.LoadPolicies(ctx, rt, compiler); err != nil {
-		return nil, err
+	if err := ruletable.LoadPolicies(ctx, protoRT, compiler); err != nil {
+		return nil, fmt.Errorf("failed to load policies: %w", err)
+	}
+
+	ruleTable, err := ruletable.NewRuleTable(index.NewMem(), protoRT)
```

Comment:
This pattern keeps repeating in almost every place that requires a rule table. Let's make it into a function and avoid the repetition.